packages/app/src/hooks/useSyncCurrentStateIntoOpenedProjects.ts

```diff
@@ -24,7 +25,7 @@ export function useSyncCurrentStateIntoOpenedProjects() {
         [currentProject.metadata.id]: {
           project: currentProject,
           fsPath: null,
-        },
+        } satisfies OpenedProjectInfo,
```

Comment:
Doesn't this already take an `OpenedProjectInfo`? How come we need all the `satisfies` now, shouldn't it already check that the value is valid? Granted this rings a bell of weird TS behavior I was running in to...

Reply:
It's actually an atom setter function, which is a callback function but the TypeScript type declarations of the setter is pretty complex. As a result, my IDE (WebStorm) was not matching usages of `OpenedProjectInfo` to this code.

So I'm not trying to enforce type here, I'm trying to fix if you "CTRL+CLICK" to find usages of `OpenedProjectInfo`. Without the `satisfies` it was hard to find where `fsPath` was being used.

I can remove these if you want, but leaving them will help with managing the code.

Yeah, I know this is a bit of pain, but I do this on my own projects to improve static typing and I can navigate around the code better and update it quicker. You only need to add them when you have things like union types which confuses IDEs.

Here is a quick example:

```
interface Cat {name: string};
interface Dog {name: string};

type Animal = Cat | Dog;

const x: Animal = {name: "example"};
```

If you try to find usages of `Cat.name` the IDE will show zero usages. The same for the `Dog.name`, because technically `x` is both and neither at the same time. (I haven't tested this example, but on larger complex types it happens).

Meanwhile, this will fix that.

```
const x: Animal = {name: "example"} satisfies Cat;
```

This is happening a lot in our code base. Making it a bit hard for me to figure out the impact of changing code.

Reply:
Interesting idea, I can see how that's useful!